### PR TITLE
Extract traversers outside methods

### PR DESCRIPTION
**Pull request by gaearon**
This makes data flow more explicit and has a side benefit of avoiding function allocations in hot methods.

### CODE REVIEW

**Comment by gaearon** [comment]
Travis seems to fail due to regenerator tests, but they were already failing before my commit on 3.0.0 branch.

File: lib/6to5/transformation/transformers/es6-let-scoping.js [hunk 1]
```diff
@@ -112,14 +112,40 @@ LetScoping.prototype.run = function () {
...
+
 /**
  * Description
  */
 
 LetScoping.prototype.remap = function () {
-  var letRefs = this.letReferences;
-  var scope   = this.scope;
-  var file    = this.file;
+  var hasRemaps = false;
```
Comment (gaearon) [inline lib/6to5/transformation/transformers/es6-let-scoping.js]: I think I got this piece by mistake [from master](https://github.com/6to5/6to5/commit/031bda61cc9a4efc29fafbe6ea8dd0e3a7ead4a4). Should I remove all changes that are present in master, but not in 3.0.0 branch?

Comment (sebmck) [inline lib/6to5/transformation/transformers/es6-let-scoping.js]: I merged `master` into `3.0.0` after this pull request so I'm not sure why it hasn't been updated.

Comment (gaearon) [inline lib/6to5/transformation/transformers/es6-let-scoping.js]: Well, at least this commit is definitely not in 3.0.0: https://github.com/6to5/6to5/commit/031bda61cc9a4efc29fafbe6ea8dd0e3a7ead4a4

It was a day ago, but master got merged into 3.0.0 two days ago.
Can you merge latest changes into 3.0.0 so I can bring my PR up to date with it?

Comment (sebmck) [inline lib/6to5/transformation/transformers/es6-let-scoping.js]: Ah, I forgot to push. Done.

Comment (gaearon) [inline lib/6to5/transformation/transformers/es6-let-scoping.js]: OK, I rebased it against 3.0.0.
Some tests are failing but those are the same ones that currently fail on 3.0.0.

**Comment by sebmck** [comment]
Thanks!

**Comment by sebmck** [comment]
I'm extremely surprised that merged with no conflicts after all the refactoring I've done...

**Comment by gaearon** [comment]
:-)

I think it's worth going over the diff once again just to make sure I made no merge mistakes.

**Comment by sebmck** [comment]
I meant my refactoring of all [this](https://github.com/6to5/6to5/tree/3.0.0/lib/6to5/transformation/transformers) that I did between you rebasing and me merging.

**Comment by gaearon** [comment]
Well, I did resolve a few conflicts updating :-)

**Pull request merged**